File: UIClient/src/main/java/cz/zelenikr/remotetouch/data/event/SmsEventContent.java
package cz.zelenikr.remotetouch.data.event;

import org.jetbrains.annotations.NotNull;

/**
 * Represents SMS.
 *
 * @author Roman Zelenik
 */
public class SmsEventContent implements EventContent {

    private final String name;

    private final String number;

    private final String content;

    private final long when;

    /**
     * @param name    Name of sender/receiver.
     * @param number  Number of sender/receiver.
     * @param content Content of SMS.
     * @param when    Timestamp of sending/receiving in milliseconds since the epoch.
     */
    public SmsEventContent(@NotNull String name, @NotNull String number, @NotNull String content, long when) {
        this.name = name;
        this.number = number;
        this.content = content;
        this.when = when;
    }

    public String getName() {
        return name;
    }

    public String getNumber() {
        return number;
    }

    public String getContent() {
        return content;
    }

    public long getWhen() {
        return when;
    }

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder("SmsEventContent{");
        sb.append("name='").append(name).append('\'');
        sb.append(", number='").append(number).append('\'');
        sb.append(", content='").append(content).append('\'');
        sb.append(", when=").append(when);
        sb.append('}');
        return sb.toString();
    }
}

File: WebClient/src/data/dto/event/SmsEventContent.kt
package data.dto.event

/**
 * Represents SMS.
 *
 * @author Roman Zelenik
 */
data class SmsEventContent(
        /**
         * Name of sender/receiver.
         */
        val name: String,
        /**
         * Number of sender/receiver.
         */
        val number: String,
        /**
         * Content of SMS.
         */
        val content: String,
        /**
         * Timestamp of sending/receiving in milliseconds since the epoch.
         */
        val time: Long
) : EventContent
